yoti-sdk-impl/src/test/java/com/yoti/api/client/spi/remote/ThirdPartyAttributeConverterTest.java

```diff
@@ -0,0 +1,125 @@
+import static org.junit.Assert.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ThirdPartyAttributeConverterTest {
+
+    private static final String SOME_ISSUANCE_TOKEN = "someIssuanceToken";
+    private static final String SOME_EXPIRY_DATE = "2019-10-15T22:04:05.123Z";
+    private static final String SOME_INVALID_FORMAT_EXPIRY_DATE = "2019-10-15";
+    private static final String SOME_DEFINITION_NAME = "com.thirdparty.id";
+
+    private static final String ENCODED_THIRD_PARTY_ATTRIBUTE = "ChFzb21lSXNzdWFuY2VUb2tlbhIvChgyMDE5LTEwLTE1VDIyOjA0OjA1LjEyM1oSEwoRY29tLnRoaXJkcGFydHkuaWQ=";
+    private static final byte[] ENCODED_THIRD_PARTY_ATTRIBUTE_BYTES = Base64.getDecoder().decode(ENCODED_THIRD_PARTY_ATTRIBUTE);
+
+    @InjectMocks
+    ThirdPartyAttributeConverter thirdPartyAttributeConverter;
+
+    @Test
+    public void shouldParseThirdPartyAttributeCorrectly() throws Exception {
+        AttributeIssuanceDetails attributeIssuanceDetails = thirdPartyAttributeConverter.parseThirdPartyAttribute(ENCODED_THIRD_PARTY_ATTRIBUTE_BYTES);
+
+        DateTime dateTimeValue = attributeIssuanceDetails.getExpiryDate();
+        String issuanceToken = attributeIssuanceDetails.getToken();
+        List<AttributeDefinition> issuingAttributes = attributeIssuanceDetails.getIssuingAttributes();
+
+        assertThat(issuingAttributes, hasSize(1));
+        assertThat(issuingAttributes.get(0).getName(), is(SOME_DEFINITION_NAME));
+        assertThat(issuanceToken, is(SOME_ISSUANCE_TOKEN));
+
+        DateTimeValueTest.assertDateTimeValue(
+                dateTimeValue, 2019, 10, 15, 22, 04, 05, 123000
+        );
+    }
+
+    @Test
+    public void shouldParseThirdPartyAttributeWithMultipleIssuingAttributes() throws Exception {
+        IssuingAttributesProto.IssuingAttributes issuingAttributes = buildIssuingAttributes(
+                SOME_EXPIRY_DATE,
+                buildDefinition("SOME_DEFINITION"),
+                buildDefinition("SOME_OTHER_DEFINITION")
+        );
+
+        byte[] thirdPartyAttribute = buildThirdPartyAttribute(SOME_ISSUANCE_TOKEN, issuingAttributes).toByteArray();
+        AttributeIssuanceDetails attributeIssuanceDetails = thirdPartyAttributeConverter.parseThirdPartyAttribute(thirdPartyAttribute);
+
+        assertThat(attributeIssuanceDetails.getIssuingAttributes().size(), is(2));
+        assertThat(attributeIssuanceDetails.getIssuingAttributes().get(0).getName(), is("SOME_DEFINITION"));
+        assertThat(attributeIssuanceDetails.getIssuingAttributes().get(1).getName(), is("SOME_OTHER_DEFINITION"));
+    }
+
+    @Test
+    public void shouldParseThirdPartyAttributeWithoutToken() throws Exception {
+        byte[] thirdPartyAttribute = buildThirdPartyAttribute("", null).toByteArray();
+        AttributeIssuanceDetails attributeIssuanceDetails = thirdPartyAttributeConverter.parseThirdPartyAttribute(thirdPartyAttribute);
+
+        assertThat(attributeIssuanceDetails.getToken(), is(""));
```

Comment:
I think we should be returning null for `attributeIssuanceDetails` here, so clients know for certain that something is amiss. Can we also check the log/error message too in this scenario?

Reply:
I spoke to @bucky-boy about checking for logs, and he said we should only do it if there was a business critical need for the log (i.e. a service that was looking for the log and triggered something else).

Reply:
Fixed in https://github.com/getyoti/yoti-java-sdk/pull/137/commits/4a8dcceb86480d32d8a4c5b5161c6f9d6caee28a